force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +136,19 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String result = '';
 
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
+        for (String name : scientistNames) {
+            // If the name contains the letter 'g' regardless of case add that name to the result string
+            if (name.containsIgnoreCase('g') == true) {
+                result += name + ','; 
+            }
+        }
         // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        result = result.removeEnd(',');
+        // Return the result string
+        return result;
```

Comment:
great work. I  love string problems. I always learn a new method. I did not know there was a containsIgnoreCase until now.